source/gui/settingsDialogs.py

```diff
@@ -114,18 +118,196 @@ def onOk(self, evt):
+		* Set L{title} to the title of the category.
+		* Override L{makeSettings} to populate a given sizer with the settings controls.
+		* Optionally, extend L{onPanelActivated} to perform actions after the category has been selected in the list of categories, such as synthesizer or braille display list population.
+		* Optionally, extend L{onPanelDeactivated} to perform actions after the category has been deselected (i.e. another category is selected) in the list of categories.
+		* Optionally, extend one or both of L{onSave} or L{onDiscard} to perform actions in response to the parent dialog's OK or Cancel buttons, respectively.
+
+	@ivar title: The title of the settings panel, also listed in the list of settings categories.
+	@type title: str
+	"""
+
+	def __init__(self, parent):
+		"""
+		@param parent: The parent for this panel; C{None} for no parent.
+		@type parent: wx.Window
+		"""
+		super(SettingsPanel, self).__init__(parent, wx.ID_ANY)
+		self.mainSizer=wx.BoxSizer(wx.VERTICAL)
+		self.settingsSizer=wx.BoxSizer(wx.VERTICAL)
+		self.makeSettings(self.settingsSizer)
+		self.mainSizer.Add(self.settingsSizer, flag=wx.ALL)
+		self.mainSizer.Fit(self)
+		self.SetSizer(self.mainSizer)
+
+	def makeSettings(self, sizer):
+		"""Populate the panel with settings controls.
+		Subclasses must override this method.
+		@param sizer: The sizer to which to add the settings controls.
+		@type sizer: wx.Sizer
+		"""
+		raise NotImplementedError
+
+	def onPanelActivated(self):
+		"""Called after the panel has been activated (i.e. de corresponding category is selected in the list of categories).
+		For example, this might be used for resource intensive tasks.
+		Sub-classes should extendthis method.
+		"""
+		self.Show()
+		self.SetSize(self.Parent.Size)
+
+	def onPanelDeactivated(self):
+		"""Called after the panel has been deactivated (i.e. another category has been selected in the list of categories).
+		Sub-classes should extendthis method.
+		"""
+		self.Hide()
+
+	def onSave(self):
+		"""Take action in response to the parent's dialog OK or apply button being pressed.
+		Sub-classes should override this method.
+		MultiCategorySettingsDialog is responsible for cleaning up the panel when OK is pressed.
+		"""
+		raise NotImplementedError
+
+	def onDiscard(self):
+		"""Take action in response to the parent's dialog Cancel button being pressed.
+		Sub-classes should override this method.
+		MultiCategorySettingsDialog is responsible for cleaning up the panel when Cancel is pressed.
+		"""
+
+class MultiCategorySettingsDialog(SettingsDialog):
+	"""A settings dialog with multiple settings categories.
+	A multi category settings dialog consists of a tree view with settings categories on the left side, 
+	and a settings panel on the right side of the dialog.
+
+	To use this dialog: set title and populate L{categoryClasses} with subclasses of SettingsPanel."""
+
+	title=""
+	categoryClasses=[]
+	settingsSizerOrientation = wx.HORIZONTAL
+	hasApplyButton = True

```

Comment:
Or perhaps the note about being inherited from `SettingsDialog` and overridden is clearer here.